### DISCUSSION TITLE
Question on SAGEConv Equation

### DISCUSSION THREAD

**Discussion by Hoonst** [pyg-team/pytorch_geometric#2101] [Q&A]
Hi, I am Recently looking into the GNN and I really appreciate your work.

As a newbie of implementing the GNN paper, I tried to use the pyG to use the SAGEConv.
But when I saw the equation on 
![image](https://user-images.githubusercontent.com/40199572/107370731-b4bac380-6b26-11eb-8902-e9c5413f305f.png)

i was not sure whether this was the equation that I saw in the paper.
![image](https://user-images.githubusercontent.com/40199572/107370910-ed5a9d00-6b26-11eb-93cf-2e890a8680d5.png)
 
As you will already know, above is the algorithm in the paper.
It describes the algorithm as simply,
1)  aggregate the neighbors
2)  concat the node with aggregation, then matrix multiplication with the Weight.
So, whether it uses GraphSAGE-GCN or Mean Aggregator, there should be 1 Weight.

But the equation you uploaded in the manual depicts that there are two weights, which I can't understand.

Can I Know the reason for this, Because I want to know whether there is anything that I missed from the paper.

Thank you!

**Comment by rusty1s** [comment]
Concatenation plus transformation
```
W @ [h_v, h_neighbors]
```
is actually equal to
```
W_1 @ h_v + W_2 @ h_neighbors
```
The latter is more memory-efficient though, as we do not need to create another node feature matrix.

**Reply by Hoonst** [reply]
Thanks for the answer!
